### PR TITLE
Add Python 3.11 to GHA

### PR DESCRIPTION
**Pull request by gliptak**

### CODE REVIEW

**Comment by gliptak** [comment]
```
  + delocate-wheel --require-archs x86_64,arm64 -w /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/repaired_wheel -v /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/built_wheel/python_snappy-0.6.1-cp38-cp38-macosx_10_9_universal2.whl
  INFO:delocate.delocating:Copying library /usr/local/Cellar/snappy/1.1.10/lib/libsnappy.1.1.10.dylib to snappy/.dylibs/libsnappy.1.1.10.dylib
  INFO:delocate.delocating:Modifying install name in snappy/_snappy.cpython-38-darwin.so from /usr/local/opt/snappy/lib/libsnappy.1.dylib to @loader_path/.dylibs/libsnappy.1.1.10.dylib
  Fixing: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/built_wheel/python_snappy-0.6.1-cp38-cp38-macosx_10_9_universal2.whl
  Traceback (most recent call last):
    File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/build/venv/bin/delocate-wheel", line 8, in <module>
      sys.exit(main())
    File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/build/venv/lib/python3.8/site-packages/delocate/cmd/delocate_wheel.py", line 128, in main
      copied = delocate_wheel(
    File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4t7bwcty/cp38-macosx_universal2/build/venv/lib/python3.8/site-packages/delocate/delocating.py", line 662, in delocate_wheel
      raise DelocationError(
  delocate.delocating.DelocationError: Some missing architectures in wheel
  Required arch arm64 missing from snappy/_snappy.cpython-38-darwin.so
```

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

**Comment by martindurant** [comment]
@gliptak , instead of updating the build and code for this package, would you perhaps consider making a PR to use the rust-based cramjam, which includes all the functionality made here? This package would become just a thin wrapper to satisfy dependencies.

**Comment by gliptak** [comment]
thank you for the tag @martindurant https://github.com/milesgranger/cramjam sounds like a bigger update

might you be open to merge/publish this PR first (assuming that I can get it to green)?

**Comment by martindurant** [comment]
> might you be open to merge/publish this PR first

Certainly, but in the meantime, py3.12 is out and 3.13 on the way, so we are already behind. 3.7 should definitely be dropped and maybe 3.8 too.

**Comment by gliptak** [comment]
https://github.com/milesgranger/cramjam/blob/master/cramjam-python/src/snappy.rs seems to offer a slightly different interface as compared to https://github.com/andrix/python-snappy/blob/master/src/snappy/__init__.py

**Comment by martindurant** [comment]
I has everything we need, no? If we can provide [stream]-de/compress, good enough I think.

**Comment by gliptak** [comment]
#127

**Comment by martindurant** [comment]
Merging from main, we no longer need to build wheels at all, this package has become noarch!

File: setup.py [hunk 1]
```diff
@@ -70,7 +70,8 @@
                  'Programming Language :: Python :: 3.7',
                  'Programming Language :: Python :: 3.8',
                  'Programming Language :: Python :: 3.9',
-                 'Programming Language :: Python :: 3.10'
+                 'Programming Language :: Python :: 3.10',
+                 'Programming Language :: Python :: 3.11'
```
Comment (martindurant) [inline setup.py:74]: + 3.12

**Comment by gliptak** [comment]
🚀

File: setup.py (hunk 1 above)
Comment (gliptak) [reply setup.py:74]: would you like to merge this `3.11` PR and `3.12` in followup PR?

**Pull request merged**